packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/text.ts

```diff
@@ -0,0 +1,22 @@
+/* ============================================================================
+ * Copyright (c) Cloud Annotations
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ * ========================================================================== */
+
+import { marked } from "marked";
+
+export function stripText(text?: string): string {
+  if (text === undefined) {
+    return "";
+  }
+  const renderer = new marked.TextRenderer();
+  marked.use({ silent: true, renderer });
+  return marked
+    .parse(text)
+    .replace(/\n/g, " ")
+    .replace(/(<([^>]+)>)/gi, " ")
+    .replace(/  +/g, " ")
+    .trim();
+}
```

Comment:
Basically what I am doing here is:

- First strip markdown formatting using `marked.TextRenderer`
- Then replace any newline characters with a space. This ensures we show `this\nthat` as `this that` instead of `thisthat`
- Then replace any HTML tags with a space and replace duplicate spaces with a single space. This allows us to replace `<p>this</p><p>that</p>` or `this<br />that` with `this that` instead of `thisthat`
- Finally trim leading/trailing whitspace

I've included test cases covering all of that.

Reply:
I’m a bit hesitant to use regex to strip the HTML since I think this would also strip out descriptions that use greater than and less than signs (I.e. “must be < 10 and > 0”).

it looks like marked allows use of a custom renderer, which doesn’t look too difficult to implement for emitting plaintext to begin with.

Reply:
Before I go and change this, just to clarify:

I'm already using marked's `TextRenderer`, which (mostly) outputs without rendering the markdown. So for example:

```js
import { marked } from "marked";
const renderer = new marked.TextRenderer();
marked.use({ silent: true, renderer });
console.log(
  marked.parse('**This** description contains [markdown](https://www.markdownguide.org/)')
)
```

outputs

```
<p>This description contains markdown</p>
```

rather than

```
<p><strong>This</strong> description contains <a href="https://www.markdownguide.org/">markdown</a></p>
```

What it _doesn't_ do is actively strip any HTML that is already in the input. So for example:

```js
import { marked } from "marked";
const renderer = new marked.TextRenderer();
marked.use({ silent: true, renderer });
console.log(
  marked.parse('<strong>This</strong> description contains <a href="https://www.w3.org/html/">HTML</a>')
)
```

outputs

```
<p><strong>This</strong> description contains <a href="https://www.w3.org/html/">HTML</a></p>
```

not

```
<p>This description contains HTML</p>
```

So if I write a custom renderer, I'm going to end up doing some version of this (stripping tags that are already present in the input using regex, or some other method) but just at a different layer.


Given that, I don't think writing a custom renderer buys us much. We're going to end up trying to strip HTML out of a string _somewhere_, so I'd probably rather write one function we apply as a post-process than do it in 9 separate functions of a custom renderer class.


If the issue is we don't want to do it with a regex, maybe it would make sense to bring in a library like https://www.npmjs.com/package/string-strip-html or https://www.npmjs.com/package/striptags instead?

Reply:
Ohh I see, I thought you were converting markdown to html and then stripping out the html 😅

Yea I’d just like to avoid regex if we can. Happy to pull in one of those libraries instead.